### ISSUE TITLE
Single group meeting audio settings

### ISSUE THREAD

**Issue by takahirox** [mozilla/hubs#2738]
This issue is top of #2662 and #2737.

If I'm right, I think Hubs is primary designed for 1. multiple groups are in a room and 2. having conversation in each group simultaneously. Audio setting is based on it, then it has positional audio and distance based audio attenuation because one doesn't want to hear other group's or far people's conversation.

Especially recent tough situations, people are more interested in remote conversation and the use cases of Hubs would be getting wider.

One of them would be Single group meeting, as we Hubs teem weekly does, where basically only one group is in a room, basically only one person speaks, and everyone wants to hear the speaker. Hubs can be used for it (as Zoom or other tools alternate).

In Single group meeting, we may need other audio settings. They are

1. No positional audio
2. No distance based audio attenuation
3. Auto members' voice volume normalization (keeping their volumes in a certain range)

For 1, we already have `/audiomode` command to enable/disable positional audio.

For 2, currently we have no way to disable distance based audio attenuation (even with disabled positional audio) #2737

For 3, currently we can manually control each person's voice volume but can't automatically do #2662

**Describe the solution you'd like**
I'd like to have options which satisfy 2 and 3 for Single group meeting. They may be an option disabling distance based audio attenuation and an option automatically normalizing all members' voice volumes. (The latter could cover the former?)

**Additional Context**
* Any other audio setting we should have for Single group meeting?
* Let me know if you think Single group meeting should not be one of our targets

**Comment by deanmax** [comment]
Maybe this is not 100% related, but I wonder if, as a room owner, that I want to announce something to the whole room, would there be some flag/mode available that I can make everyone in the room hear me?

**Comment by robertlong** [comment]
I think we should really think a lot about the design of these things. The first pass at #2 with the `/rolloff` command seems like a ok interim solution, but that and the `/audiomode` command do not strike me as good user-oriented design.

Here are some ideas of things we could do:

### 1. Temporary User Audio Settings
Adjusting audio settings to disable/adjust audio spatialization in a single room is somewhat heavy-handed. If a user turns off audio spatialization or turns up audio rolloff for one room, they may forget about it in their next visit to Hubs and not know what's happening. Perhaps there's some additional UI that we could add to the HUD, separate from the preference pane that had these temporary audio settings. I could imagine a checkbox for disabling positional audio and a slider for adjusting audio rolloff with visual feedback in the room to show you where you will stop hearing media and users. The next time you visit Hubs these settings will be reset.

### 2. Room Audio Settings
In a meeting with a lot of people, such as our standups, it can be hard to hear people when they are far away. In these situations, it may be better to just adjust these settings for all users in the room. When creating a room you could select an option to have audio spatialization disabled for all users. These settings would also be available to you once you are inside the room in the room settings dialog. Now only the room creator/moderators need to adjust these settings for their users.

### 3. Scene Triggered Audio Settings (Audio Zones)
In a conference/meetup use-case you want to be able to isolate conversaions/talks from one another and be able to hear people talking clearly. Audio zones would provide a visual cue that this is a space in a meeting room where audio spatialization will be disabled. The cue could be a semi-transparent volume with a speaker icon stamped on the sides. You could also show this status in the HUD. Audio Zones would be placed in Spoke. I think by default, this would turn off both positional audio and distance based audio attenuation, but maybe those settings would be exposed as advanced settings. Perhaps there's also a way to designate speaker controls when inside this volume. The room owner/moderators can pick who can talk in this space and maybe enable/disable it on the fly so that users can ask/answer questions.

### 4. Moderator Voice Broadcasting (Intercom Tool)
When a room moderator is trying to talk to everyone in the room but audio spatialization is turned on, it can be hard to get everyone's attention. Text chat is ok, but easy to ignore. In this case the intercom tool can be used to talk to everyone. As a room moderator, you hold down the intercom button in the ui or the hotkey and you can speak to everyone in the room. On their side, all media volume quickly lowers so you mainly hear what the moderator is saying, kinda like the intercom on an airplane.

**Comment by takahirox** [comment]
Agreed with improving the design. I'm thinking of adding room type option which room owner/moderator can select and has an effect to all members in a room, for example "meeting" where positional audio and audio rolloff are disabled and "conference" where particular members'/audios' volumes are louder and/or their positional audio and audio rolloff are disabled. And it may be good to add advanced options for owners who want to tune audio more fine, for example adjusting each person's volume, rolloff, or etc (room type can be said as presets of advanced options).

For me, `/audioRolloff` command #2766 is not the goal for this issue. But it's kind of experimental feature to see how good/bad the experience of disabling audio rolloff. (And I guess most of the core would be reusable even if we change the design).

BTW, regarding "Temporary User Audio Settings", already `audiomode` and `audioRolloff` are reset when a user enters a room.
